pkg/util/machines.go

```diff
@@ -114,12 +114,24 @@ func IsControlPlaneMachine(machine metav1.Object) bool {
 // GetMachineMetadata returns the cloud-init metadata as a base-64 encoded
 // string for a given VSphereMachine.
 func GetMachineMetadata(hostname string, vsphereVM infrav1.VSphereVM, networkStatuses ...infrav1.NetworkStatus) ([]byte, error) {
+	return GetMachineMetadataWithIPAMState(hostname, vsphereVM, map[string]infrav1.NetworkDeviceSpec{}, networkStatuses...)
```

Comment:
I don't think we need rename this. ultimately this function is supposed to return the metadata of a VM, we shouldn't leak the content into the name of the function

Reply:
We made a second function to preserve the original function. `make apidiff` fails when we add a parameter to the original function. We can add the parameter to the original function name, but that means changing the 'public' api. 
Is it acceptable to submit this PR with failing apidiff?

Reply:
yeah we're using it mostly as a signal to know what we're breaking, we're not providing contract stability for functions